ts/components/Pinpad/index.tsx

```diff
@@ -72,11 +75,33 @@ class Pinpad extends React.PureComponent<Props, State> {
     ]
   ];
 
+  private shake() {
+    // disable click keypad
+    this.setState({
+      disabled: true
+    });
+    // enable after 1 second
+    setTimeout(() => {
+      this.setState({
+        disabled: false
+      });
+    }, 1000);
+    this.animatedValue.setValue(0);
+    // start animation
+    Animated.timing(this.animatedValue, {
```

Comment:
Instead of using the animated value to control the margin, check if you can shake the component using transform/ translate. The reason is that with transform you can use the native driver.

Reply:
If you can don't forget to add useNativeDriver: true here.

Reply:
ok I'll try

Reply:
I think I did everything, I turned the shake from vertical to horizontal because it seemed more pleasant